### PR TITLE
added associations

### PR DESCRIPTION
**Pull request by Samuelwanza**
I have conducted the following:
-Created a class Classroom with the following:
@label instance variable, should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
-Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
-Created the has-many side (a classroom has many students).
-Created the belongs-to side (a student belongs to a classroom).
-Made sure that when adding a student to a classroom it also sets the classroom for the student.
-Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
-Created a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
-Created a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
-Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
-Created the has-many side of Book and Rental (a book has many rentals).
-Created the belongs-to side of Rental and Book (a rental belongs to a book).
-Created the has-many side of Person and Rental (a person has many rentals).
-Created the belongs-to side of Rental and Person (a rental belongs to a person).
-Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -6,9 +6,10 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (tufoinnkuo10) [inline person.rb:13]: - [ ] please kindly ensure there's a method for adding rentals here as well. Here, The person itself should be sent as a parameter to create the new rental, achieving the 'has-many' association`
something like
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+  attr_accessor :title, :author
+  attr_reader :rentals
+end
```
Comment (tufoinnkuo10) [inline book.rb:13]: - [ ] great job so far. please kindly ensure that a method for adding rentals should be present. note that here the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.`
eg
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Samuelwanza 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally


Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply person.rb:13]: Fixed.

File: book.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply book.rb:13]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Samuelwanza 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**